bug: test_startup_time fails for azure

**What happened:**
- platform test test_startup_time fails on azure platform (see debug logs below)
- test is executing `systemd-analyze` before systemd finished booting 

**Debug Logs:**
```
>       assert exit_code == 0, f"no {error=} expected"
E       AssertionError: no error="Bootup is not yet finished (org.freedesktop.systemd1.Manager.FinishTimestampMonotonic=0).\nPlease try again later.\nHint: Use 'systemctl list-jobs' to see active jobs\n" expected
E       assert 1 == 0
```
https://github.com/gardenlinux/gardenlinux/actions/runs/4988419582/jobs/8931721990#step:11:584

--- a/tests/helper/utils.py
+++ b/tests/helper/utils.py
@@ -116,6 +116,16 @@ def get_kernel_version(client):
     return output
 
 
+def wait_systemd_boot(client):
+    """ Wait for systemd to finish booting """
+
+    cmd = "systemctl is-system-running --wait"
+
+    (exit_code, output, error) = client.execute_command(cmd, quiet=False)
+
+    assert exit_code == 0, f"Failed to wait for systemd"
+
+
 def validate_systemd_unit(client, systemd_unit, active=True):
     """ Validate a given systemd unit """
     cmd = f"systemctl status {systemd_unit}.service"

--- a/tests/integration/azure.py
+++ b/tests/integration/azure.py
@@ -4,6 +4,7 @@ import re
 import pytest
 import uuid
 
+from helper.utils import wait_systemd_boot
 from helper.sshclient import RemoteClient
 
 from azure.core.exceptions import (
@@ -434,6 +435,8 @@ class AZURE:
                 host=ip.ip_address,
                 sshconfig=config["ssh"],
             )
+            logger.info("Waiting for systemd to finish with boot")
+            wait_systemd_boot(ssh)
             yield ssh
         finally:
             if ssh is not None:

